// hypixel.ts/dist/rest/APIRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIRequest = void 0;
const undici_1 = require("undici");
const errors_1 = require("../errors");
class APIRequest {
    rest;
    method;
    path;
    options;
    route;
    client;
    constructor(rest, method, path, options) {
        this.rest = rest;
        this.method = method;
        this.path = path;
        this.options = options;
        this.route = options.route;
        this.client = rest.client;
        if (options.query && typeof options.query === 'object') {
            const queryString = Object.entries(options.query)
                .filter(([, value]) => value !== null && typeof value !== 'undefined')
                .map(([key, value]) => (Array.isArray(value) ? `${key}=${value.join(',')}` : `${key}=${value}`))
                .join('&');
            this.path = `${path}?${queryString}`;
        }
    }
    async make() {
        const { apiKey } = this.client;
        const headers = {};
        if (!apiKey)
            throw new errors_1.HypixelTSError('CLIENT_OPTION_MISSING', 'apiKey');
        headers['API-Key'] = apiKey;
        let body = null;
        if (this.method !== 'get' && this.options.body) {
            body = JSON.stringify(this.options.body);
            headers['Content-Type'] = 'application/json';
        }
        return (0, undici_1.request)(`https://api.hypixel.net${this.path}`, {
            body: body,
            headers
        });
    }
}
exports.APIRequest = APIRequest;
