// hypixel.ts/dist/lib/Client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
const rest_1 = require("../rest");
const HypixelTSError_1 = require("../errors/HypixelTSError");
const managers_1 = require("../managers");
const classes_1 = require("../classes");
class Client {
    constructor(apiKey) {
        if (!apiKey)
            throw new HypixelTSError_1.HypixelTSError('CLIENT_OPTIONS_MISSING', 'apiKey');
        if (typeof apiKey !== 'string')
            throw new HypixelTSError_1.HypixelTSError('CLIENT_OPTION_INVALID_TYPE', 'apiKey', 'string');
        this.apiKey = apiKey;
        this.registerManagers();
    }
    get api() {
        return this.rest.routeBuilder;
    }
    registerManagers() {
        this.rest = new rest_1.RestManager(this);
        this.players = new managers_1.PlayerManager(this);
        this.skyblock = new managers_1.SkyBlockManager(this);
        this.resources = new managers_1.ResourcesManager(this);
        this.other = new managers_1.OtherManager(this);
        this.guilds = new managers_1.GuildManager(this);
        this.util = new classes_1.Util(this);
    }
}
exports.Client = Client;
