// hypixel.ts/dist/managers/OtherManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OtherManager = void 0;
const _1 = require(".");
class OtherManager extends _1.BaseManager {
    constructor(client) {
        super(client);
    }
    async getActiveBoosters() {
        const { boosters, boosterState } = await this.client.api.boosters.get();
        return { boosters, boosterState };
    }
    async getPlayerCount() {
        const { playerCount, games } = await this.client.api.counts.get();
        return { playerCount, games };
    }
    async getLeaderboard() {
        const { leaderboards } = await this.client.api.leaderboards.get();
        return leaderboards;
    }
    async getPunishmentStatistics() {
        const { watchdog_lastMinute, staff_rollingDaily, watchdog_total, watchdog_rollingDaily, staff_total } = await this.client.api.punishmentstats.get();
        return { watchdog_lastMinute, staff_rollingDaily, watchdog_total, watchdog_rollingDaily, staff_total };
    }
    async getAPIKeyInformation() {
        return (await this.client.api.key.get()).record;
    }
}
exports.OtherManager = OtherManager;
