// hypixel.ts/dist/managers/ResourcesManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourcesManager = void 0;
const _1 = require(".");
class ResourcesManager extends _1.BaseManager {
    constructor(client) {
        super(client);
    }
    async gameInfo() {
        const { lastUpdated, games } = await this.client.api.resources.games.get();
        return { lastUpdated, games };
    }
    async getAchievements() {
        const { lastUpdated, achievements } = await this.client.api.resources.achievements.get();
        return { lastUpdated, achievements };
    }
    async getChallenges() {
        const { lastUpdated, challenges } = await this.client.api.resources.challenges.get();
        return { lastUpdated, challenges };
    }
    async getQuests() {
        const { lastUpdated, quests } = await this.client.api.resources.quests.get();
        return { lastUpdated, quests };
    }
    async getGuildAchievements() {
        const { lastUpdated, one_time, tiered } = await this.client.api.resources.guilds.achievements.get();
        return { lastUpdated, one_time, tiered };
    }
    async getVanityPets() {
        const { lastUpdated, types, rarities } = await this.client.api.resources.vanity.pets.get();
        return { lastUpdated, types, rarities };
    }
    async getVanityCompanions() {
        const { lastUpdated, types, rarities } = await this.client.api.resources.vanity.companions.get();
        return { lastUpdated, types, rarities };
    }
}
exports.ResourcesManager = ResourcesManager;
